### PR TITLE
RoundedHighlighter for components with rounded corners

### PR DESCRIPTION
**Pull request by Taikelenn**
# Description of the PR
Before:
obraz
After:
obraz

# Testing
<table>
  <tr>
    <th>unit</th>
    <th>integration</th>
    <th>manual</th>
  </tr>
  <tr>
    <td>
      <ul>
        <li>- [ ] new <b>unit</b> tests created</li>
        <li>- [X] all <b>unit</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>integration</b> tests created</li>
        <li>- [X] all <b>integration</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [X] <b>manual</b> testing went well</li>
      </ul>
    </td>
  </tr>
</table>  
  
# Have you updated the [TESTING.md](https://github.com/Aalto-LeTech/aplus-courses/blob/master/TESTING.md) or other relevant documentation on _your_ branch?

- [ ] Yes.
- [ ] Not yet. I will do it next.
- [x] Not relevant.

### CODE REVIEW

**Review by jaakkonarhi — APPROVED** [review]
Thank you very much

File: src/main/java/fi/aalto/cs/apluscourses/ui/ideactivities/OverlayPane.java [hunk 1]
```diff
@@ -93,6 +94,9 @@ protected void paintComponent(Graphics graphics) {
     Graphics2D g = (Graphics2D) graphics.create();
     g.setComposite(AlphaComposite.SrcOver.derive(0.7f));
     g.setColor(Color.BLACK); // using JBColor.BLACK is wrong here
+    g.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
+    g.setRenderingHint(RenderingHints.KEY_RENDERING, RenderingHints.VALUE_RENDER_QUALITY);
+    g.setRenderingHint(RenderingHints.KEY_INTERPOLATION, RenderingHints.VALUE_INTERPOLATION_BILINEAR);
```
Comment (OlliKiljunen) [inline src/main/java/fi/aalto/cs/apluscourses/ui/ideactivities/OverlayPane.java:99]: Is it known if these may hit the performance on less efficient machines?

File: src/main/java/fi/aalto/cs/apluscourses/ui/ideactivities/RoundedHighlighter.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import java.awt.geom.RoundRectangle2D;
+import java.util.Collections;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+
+public class RoundedHighlighter extends GenericHighlighter {
+
+  @Override
+  public List<RectangularShape> getArea() {
+    final float arc = new JBValue.UIInteger("Button.arc", 6).getFloat();
```
Comment (OlliKiljunen) [inline src/main/java/fi/aalto/cs/apluscourses/ui/ideactivities/RoundedHighlighter.java:15]: Could this be a class field (perhaps even static)?

**Review by OlliKiljunen — APPROVED** [review]
Very good PR! It also seems that the underlying design works well, because this extension could be so nicely done. 👍 

I wrote a couple of questions which you may want to consider. Anyway, I'm fine with this as-is.

File: src/main/java/fi/aalto/cs/apluscourses/ui/ideactivities/OverlayPane.java (hunk 1 above)
Comment (jaakkonarhi) [reply src/main/java/fi/aalto/cs/apluscourses/ui/ideactivities/OverlayPane.java:99]: IntelliJ uses these too so maybe it's not a problem

Comment (Taikelenn) [reply src/main/java/fi/aalto/cs/apluscourses/ui/ideactivities/OverlayPane.java:99]: Less efficient machines will not run IntelliJ. :>
In all seriousness, it won't affect performance in any noticeable way.

File: src/main/java/fi/aalto/cs/apluscourses/ui/ideactivities/RoundedHighlighter.java (hunk 2 above)
Comment (Taikelenn) [reply src/main/java/fi/aalto/cs/apluscourses/ui/ideactivities/RoundedHighlighter.java:15]: Yes, but if the user changes the theme during a tutorial (for some obscure reason) then a static field won't get updated :<

**Pull request merged**